Fix for a hangup when viewing a PVR recording and pressing next/previ…

hugegreenbug opened the pull request:
…ous (#17476)

<!--- Provide a general summary of your change in the Title above -->
This fix skips channel up/down events for saved recordings to prevent the hang

## Description
<!--- Describe your change in detail -->
Fixes a hang when pressing previous or next while viewing a saved recording

http://trac.kodi.tv/ticket/17476

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here -->

## How Has This Been Tested?
<!--- Please describe in detail how you tested your change -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc -->
Compiled on MacOS. Binary addons are not showing up with the latest master, so I haven't been able to test fully.

## Screenshots (if appropriate):

## Types of change
<!--- What type of change does your code introduce? Put an `x` in all the boxes that apply like this: [X] -->
- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply like this: [X] -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [X ] My code follows the [Code guidelines](https://codedocs.xyz/xbmc/xbmc/code_guidelines.html) of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [X ] I have read the [CONTRIBUTING](https://github.com/xbmc/xbmc/blob/master/CONTRIBUTING.md) document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

ksooo commented:
I bet that you have not tested this, because it will not even compile. In master, there is no symbol g_PVRManager anymore. Come on, you can do better.

Hint: CServiceBroker::GetPVRManager()

hugegreenbug commented:
Yeah, I said I didn't test it yet. I'm working on it.

ksooo commented:
> Yeah, I said I didn't test it yet. I'm working on it.

 It is no good practice to submit a PR that has not been compiled/tested locally on at least one platform before. Submitting a PR draws attention of other people and I hate (!) spending my time reviewing a PR and detecting syntax errors you know. This is the job of compilers before a human beeing gets bothered with the code. ;-)

hugegreenbug commented:
OK, sorry.

On Jun 1, 2017 5:00 AM, "Kai Sommerfeld" <notifications@github.com> wrote:

> Yeah, I said I didn't test it yet. I'm working on it.
>
> It is no good practice to submit a PR that has not been compiled/tested
> locally on at least one platform before. Submitting a PR draws attention of
> other people and I hate (!) spending my time reviewing a PR and detecting
> syntax errors you know. This is the job of compilers before a human beeing
> gets bothered with the code. ;-)
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/xbmc/xbmc/pull/12210#issuecomment-305460813>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/ABlFSe5Ka6ky4i27r6h6RLTuZ6smyFp-ks5r_pm1gaJpZM4NsxW5>
> .
>

ksooo commented:
Nono, let's keep it open now that things are sorted out. My goal was to engage you to do better *next* time. ;-)

pollito05 commented:
yes

hugegreenbug commented:
@ksooo I have changed g_PVRManager to CServiceBroker::GetPVRManager() and compiled it locally. I am unable to test fully since the binary addons are not showing up in my build. I was told it may be due to recent changes.

ksooo commented:
@hugegreenbug can you please squash changes into a single commit.

hugegreenbug committed:
Fix for a hangup when viewing a PVR recording and pressing next/previous
 - This fix skips channel up/down events for saved recordings to prevent the hang
 - Bug #17476

hugegreenbug commented:
@ksooo Ok, I committed it as a single commit.

ksooo commented:
Thank you

jenkins build and merge
